### ISSUE TITLE
Passing a null arg triggers defaultValue error with Typescript stories

### ISSUE THREAD

**Issue by derek-duncan** [storybookjs/storybook#12191]
**Describe the bug**
This was an odd bug and difficult to track down. When you pass `null` to one of your `args` in CSF format, it triggers an error `Cannot read property 'defaultValue' of undefined`. If I remove these `null` props, then the error goes away. I suspect this is an issue with Storybook trying to create an args table automatically from Typescript files.

**To Reproduce**
```
function Test(props: {text: string | null}) {
  return <span>{props.text}</span>;
}

export default {
  title: 'Test',
  component: Test,
  args: {
    text: null,
  },
} as Meta;

export const Default = (args) => <Test {...args} />;
```

**Expected behavior**
`null` values should be supported.

**Screenshots**
<img width="1179" alt="Dose_Calculator_-_Both_Calculators_⋅_Storybook" src="https://user-images.githubusercontent.com/2983822/90941046-96290a80-e3d6-11ea-91fa-5e7a44597c5b.png">

**Code snippets**
If applicable, add code samples to help explain your problem.
```
TypeError: Cannot read property 'defaultValue' of undefined
    at http://localhost:9001/vendors~main.0731b9592eed5f3eeb81.bundle.js:5886:38
    at Array.reduce (<anonymous>)
    at StoryStore.addStory (http://localhost:9001/vendors~main.0731b9592eed5f3eeb81.bundle.js:5883:50)
    at Object.api.add (http://localhost:9001/vendors~main.0731b9592eed5f3eeb81.bundle.js:4268:25)
    at http://localhost:9001/vendors~main.0731b9592eed5f3eeb81.bundle.js:7282:16
    at Array.forEach (<anonymous>)
    at http://localhost:9001/vendors~main.0731b9592eed5f3eeb81.bundle.js:7256:20
    at Array.forEach (<anonymous>)
    at http://localhost:9001/vendors~main.0731b9592eed5f3eeb81.bundle.js:7189:11
    at ConfigApi.configure (http://localhost:9001/vendors~main.0731b9592eed5f3eeb81.bundle.js:4366:7)
```

**Additional context**
Using `@storybook/react@^6.0.16`. No addons.

**Comment by trevdor** [comment]
@shilman I was able to repro here: 
https://codesandbox.io/s/storybook-defaultvalue-error-xw01u?file=/src/stories/Magic.stories.tsx

The error is thrown on line 416 here:
https://github.com/storybookjs/storybook/blob/0bf34e0ecc90020fb3860b566070f96f122f94dd/lib/client-api/src/story_store.ts#L411-L418

Seems to come from having an arg with no corresponding argType configured, such that `argTypes[arg]` would be undefined.

My commented bit at the top of that sandbox addresses the problem, but it appears `argTypes` are meant to be optional.

**Comment by webholics** [comment]
I have the same issue when disabling the docs extension. I'm doing this currently because of performance problems with the docs extension which leads to very slow storybook init times.

**Comment by shilman** [comment]
closing as dupe to #12732 and fixed in 6.1-beta

```
npx sb upgrade --prerelease
```
